File: 2024.2/aaa-scc0218/guloso1.py
def trocas(v, n):
    rounds = 0
    target = 1

    while target <= n:
        rounds += 1
        for x in v:
            if target == x:
                target += 1
    print(rounds)


n = int(input())
vetor = list(map(int, input().strip().split()))

trocas(vetor, n)

File: 2024.2/aaa-scc0218/guloso1/guloso1.c
#include <stdbool.h>
#include <stdio.h>

void trocas(int v[], int n) {
    int rounds = 0;
    bool found[n + 1];

    // Initialize the boolean array
    for (int i = 0; i <= n; i++) {
        found[i] = false;
    }

    int count = 0;

    while (count < n) {
        rounds++;
        for (int i = 0; i < n; i++) {
            if (!found[v[i]] && v[i] == count + 1) {
                found[v[i]] = true;
                count++;
            }
        }
    }

    printf("%d\n", rounds);
}

int main() {
    int n;
    scanf("%d", &n);

    int vetor[n];
    for (int i = 0; i < n; i++) {
        scanf("%d", &vetor[i]);
    }

    trocas(vetor, n);

    return 0;
}
